strutil/pathiter.go

```diff
@@ -58,48 +58,72 @@ func NewPathIterator(path string) (*PathIterator, error) {
 	return &PathIterator{path: path}, nil
 }
 
-// Path returns the path being traversed.
+// Path returns the entire path being traversed.
 func (iter *PathIterator) Path() string {
 	return iter.path
 }
 
-// CurrentName returns the name of the current path element.
-// The return value may end with '/'. Use CurrentNameNoSlash to avoid that.
-func (iter *PathIterator) CurrentName() string {
-	return iter.path[iter.left:iter.right]
-}
-
-// CurrentNameNoSlash returns the same value as Name with right slash trimmed.
-func (iter *PathIterator) CurrentNameNoSlash() string {
-	if iter.right > 0 && iter.path[iter.right-1:iter.right] == "/" {
+// CurrentBase returns the name of the current path element.
+// It will only end with a slash if the iterator is currently
+// placed at the base of the filesystem.
+// Ex:
+//
+//	iter:= NewPathIterator("/foo/bar")
+//	iter.Next() // iter.CurrentBase() == "/" && iter.CurrentPath() == "/"
+//	iter.Next() // iter.CurrentBase() == "foo" && iter.CurrentPath() == "/foo"
+//	iter.Next() // iter.CurrentBase() == "bar" && iter.CurrentPath() == "/foo/bar"
+func (iter *PathIterator) CurrentBase() string {
+	// Only remove ending slash if the iterator is not pointing to the base of the filesystem
+	if iter.right > 2 && iter.path[iter.right-1:iter.right] == "/" {
 		return iter.path[iter.left : iter.right-1]
 	}
 	return iter.path[iter.left:iter.right]
 }
 
-// CurrentPath returns the prefix of path that was traversed, including the current name.
+// CurrentPath returns the prefix of path that was traversed, including the current base.
 func (iter *PathIterator) CurrentPath() string {
-	return iter.path[:iter.right]
-}
-
-// CurrentPathNoSlash returns the same value as CurrentPath with the right slash trimmed.
-func (iter *PathIterator) CurrentPathNoSlash() string {
-	if iter.right > 0 && iter.path[iter.right-1:iter.right] == "/" && iter.path[:iter.right] != "/" {
+	// Only remove ending slash if the iterator is not pointing to the base of the filesystem
+	if iter.right > 2 && iter.path[iter.right-1:iter.right] == "/" {
```

Comment:
why the change to 2 here?

Reply:
That was a mistype on my part. I meant to change it to "1" so that if the only part of the CurrentPath is the base of the filesystem, that slash does not get removed. I made that change to conform to path/filepath where a path of only a "/" is considered a clean path.